src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/RecordAudio/RecordAudio.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Box, Button } from '@mui/material';
+
+import { postFileUploadApi } from 'api';
+import { StyledLinearProgress, StyledTitleMedium, theme } from 'shared/styles';
+import { Modal, Svg } from 'shared/components';
+import { useAsync } from 'shared/hooks';
+import { getUploadFormData } from 'shared/utils';
+import { MLPlayer } from 'modules/Builder/components';
+
+import { useAudioRecorder } from './RecordAudio.hooks';
+import { RecordAudioProps } from './RecordAudio.types';
+import { StyledButtons, StyledRecordButton } from './RecordAudio.styles';
+import { formatSecondsToMinutes } from './RecordAudio.utils';
+
+export const RecordAudio = ({ open, onUpload, onChange, onClose }: RecordAudioProps) => {
+  const { t } = useTranslation('app');
+  const [url, setUrl] = useState('');
+
+  const {
+    startRecording,
+    stopRecording,
+    togglePauseResume,
+    clearBlob,
+    recordingBlob,
+    isRecording,
+    isPaused,
+    isStopped,
+    recordingTime,
+  } = useAudioRecorder();
+
+  const { execute: executeAudioUpload, isLoading } = useAsync(
+    postFileUploadApi,
+    (response) => response?.data?.result && setUrl(response?.data?.result.url ?? ''),
```

Comment:
I think you don't need conditional chain in the setUrl, maybe only for url:
(response) => response?.data?.result && setUrl(response.data.result?.url ?? ''),

Reply:
fixed